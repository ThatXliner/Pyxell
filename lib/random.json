{
	"node": "Block",
	"position": [
		2,
		1
	],
	"stmts": [
		{
			"node": "StmtFunc",
			"position": [
				2,
				1
			],
			"id": "seed",
			"typevars": [],
			"args": [
				{
					"node": "FuncArg",
					"position": [
						2,
						11
					],
					"name": "x",
					"type": {
						"node": "TypeName",
						"position": [
							2,
							14
						],
						"name": "Int"
					},
					"default": null
				}
			],
			"ret": {
				"node": "TypeName",
				"position": [
					2,
					20
				],
				"name": "Void"
			},
			"block": null
		},
		{
			"node": "StmtFunc",
			"position": [
				4,
				1
			],
			"id": "randInt",
			"typevars": [],
			"args": [
				{
					"node": "FuncArg",
					"position": [
						4,
						14
					],
					"name": "r",
					"type": {
						"node": "TypeName",
						"position": [
							4,
							17
						],
						"name": "Int"
					},
					"default": {
						"node": "AtomInt",
						"position": [
							4,
							23
						],
						"value": 2
					}
				}
			],
			"ret": {
				"node": "TypeName",
				"position": [
					4,
					27
				],
				"name": "Int"
			},
			"block": null
		},
		{
			"node": "StmtFunc",
			"position": [
				5,
				1
			],
			"id": "randFloat",
			"typevars": [],
			"args": [
				{
					"node": "FuncArg",
					"position": [
						5,
						16
					],
					"name": "r",
					"type": {
						"node": "TypeName",
						"position": [
							5,
							19
						],
						"name": "Float"
					},
					"default": {
						"node": "AtomInt",
						"position": [
							5,
							27
						],
						"value": 1
					}
				}
			],
			"ret": {
				"node": "TypeName",
				"position": [
					5,
					31
				],
				"name": "Float"
			},
			"block": null
		}
	]
}