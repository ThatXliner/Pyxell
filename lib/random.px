
func seed(Int s) extern

func randInt(Int r: 2) Int extern
func randFloat(Float r: 1) Float extern
