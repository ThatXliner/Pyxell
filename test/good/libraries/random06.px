
use random

for i in 0...2 do
    seed(-3456)
    print randFloat()
    print randFloat()
