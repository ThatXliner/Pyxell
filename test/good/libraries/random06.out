0.365676659974534
0.89382462424748
0.365676659974534
0.89382462424748
